packages/aws-cdk-lib/aws-stepfunctions/lib/state-machine.ts

```diff
@@ -507,26 +509,32 @@ export class StateMachine extends StateMachineBase {
   }
 
   private buildLoggingConfiguration(logOptions: LogOptions): CfnStateMachine.LoggingConfigurationProperty {
-    // https://docs.aws.amazon.com/step-functions/latest/dg/cw-logs.html#cloudwatch-iam-policy
-    this.addToRolePolicy(new iam.PolicyStatement({
-      effect: iam.Effect.ALLOW,
-      actions: [
-        'logs:CreateLogDelivery',
-        'logs:GetLogDelivery',
-        'logs:UpdateLogDelivery',
-        'logs:DeleteLogDelivery',
-        'logs:ListLogDeliveries',
-        'logs:PutResourcePolicy',
-        'logs:DescribeResourcePolicies',
-        'logs:DescribeLogGroups',
-      ],
-      resources: ['*'],
-    }));
+    if (logOptions.level !== LogLevel.OFF && !logOptions.destination) {
+      throw new Error('Logs destination is required when level is not OFF.');
+    }
+
+    if (logOptions.destination) {
+      // https://docs.aws.amazon.com/step-functions/latest/dg/cw-logs.html#cloudwatch-iam-policy
+      this.addToRolePolicy(new iam.PolicyStatement({
+        effect: iam.Effect.ALLOW,
+        actions: [
+          'logs:CreateLogDelivery',
+          'logs:GetLogDelivery',
+          'logs:UpdateLogDelivery',
+          'logs:DeleteLogDelivery',
+          'logs:ListLogDeliveries',
+          'logs:PutResourcePolicy',
+          'logs:DescribeResourcePolicies',
+          'logs:DescribeLogGroups',
+        ],
+        resources: ['*'],
+      }));
+    }
 
     return {
-      destinations: [{
+      destinations: logOptions.destination ? [{
         cloudWatchLogsLogGroup: { logGroupArn: logOptions.destination.logGroupArn },
-      }],
+      }] : undefined,
```

Comment:
if we do this like as follows - will it reduce need to check `logOptions.destination` twice and simplify a bit?
```
let destinations = undefined;
if (logOptions.destination) {
  // Policy addition
  destinations = [{
    cloudWatchLogsLogGroup: { logGroupArn: logOptions.destination.logGroupArn },
  }];
}

return {
  destinations,
  ...,
  level: logOptions.level || LogLevel.ERROR
}
```

Reply:
updated.